homeassistant/components/heatzy/climate.py

```diff
@@ -0,0 +1,294 @@
+            ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            | ClimateEntityFeature.PRESET_MODE
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return current temperature."""
+        cur_temp_high = self._conf_attr.get(CUR_TEMP_H, 0)
+        cur_temp_low = self._conf_attr.get(CUR_TEMP_L, 0)
+        return (cur_temp_low + (cur_temp_high * 255)) / 10
+
+    @property
+    def target_temperature_high(self) -> float:
+        """Return comfort temperature."""
+        cft_temp_high = self._conf_attr.get(CFT_TEMP_H, 0)
+        cft_temp_low = self._conf_attr.get(CFT_TEMP_L, 0)
+        return (cft_temp_low + (cft_temp_high * 255)) / 10
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Return comfort temperature."""
+        eco_temp_high = self._conf_attr.get(ECO_TEMP_H, 0)
+        eco_temp_low = self._conf_attr.get(ECO_TEMP_L, 0)
+        return (eco_temp_low + (eco_temp_high * 255)) / 10
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temp_eco = kwargs[ATTR_TARGET_TEMP_LOW]
+        temp_cft = kwargs[ATTR_TARGET_TEMP_HIGH]
+
+        if (temp_eco or temp_cft) is None:
+            return
+
+        self._conf_attr[ECO_TEMP_L] = int(temp_cft * 10)
+        self._conf_attr[CFT_TEMP_L] = int(temp_eco * 10)
+
+        try:
+            await self.coordinator.api.async_control_device(
+                self.unique_id,
+                {
+                    CONF_ATTRS: {
+                        CFT_TEMP_L: self._conf_attr[CFT_TEMP_L],
+                        ECO_TEMP_L: self._conf_attr[ECO_TEMP_L],
+                    }
+                },
+            )
+            await self.coordinator.async_request_refresh()
+        except HeatzyException as error:
+            _LOGGER.error(
+                "Error to set temperature: %s",
+                error,
+                exc_info=True,
+            )
+
+    async def async_turn_on(self) -> None:
+        """Turn device on."""
+        try:
+            await self.coordinator.api.async_control_device(
+                self.unique_id, {CONF_ATTRS: {CONF_ON_OFF: 1}}
+            )
+            await self.coordinator.async_request_refresh()
+        except HeatzyException as error:
+            _LOGGER.error(
+                "Error to turn on : %s",
+                error,
+                exc_info=True,
+            )
+
+    async def async_turn_off(self) -> None:
+        """Turn device off."""
+        try:
+            await self.coordinator.api.async_control_device(
+                self.unique_id, {CONF_ATTRS: {CONF_ON_OFF: 0}}
+            )
+            await self.coordinator.async_request_refresh()
+        except HeatzyException as error:
+            _LOGGER.error(
+                "Error to turn off : %s",
+                error,
+                exc_info=True,
+            )
+
+    @property
+    def hvac_mode(self) -> HVACMode:
```

Comment:
Is this overriding the superclass's `hvac_mode` property on purpose?

Reply:
Effectively it overloads the parent class